.github/workflows/build-and-publish.yaml

```diff
@@ -12,9 +12,63 @@ env:
   GHOSTSCRIPT_URL: https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs10030/gs_10.03.0_amd64_snap.tgz
   PO4A_GH_URL: https://github.com/mquinson/po4a/releases/download
   PO4A_VERSION: 0.73
+  IMAGEMAGICK_URL: https://github.com/ImageMagick/ImageMagick/releases/download/7.1.1-46/ImageMagick-8209e84-gcc-x86_64.AppImage
 
 jobs:
+  make_monochrome:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'pull_request'
+
+    steps:
+      - name: Set up repository
+        uses: actions/checkout@v4
+
+      - name: Install magick
+        run: |
+          sudo add-apt-repository universe
+          sudo apt-get install -qq -y --no-install-recommends libfuse2t64  # FUSE is required for GH runners to run AppImages
+          wget --quiet ${{ env.IMAGEMAGICK_URL }}
+          find . -name "ImageMagick*" -exec mv '{}' magick \;
+          chmod a+x magick
+
+      - name: Adjust map files for monochrome buld
+        run: |
+          find . -type f -name "*tex" -not -regex ".*/\(draft-scenarios\|translated\|svg-inkscape\|templates\)/.*" \
+            -exec grep -o "maps.*png" '{}' \; | while IFS= read -r IMG; do
+            IMG="assets/${IMG}"
+            echo "Converting ${IMG}"
+            TEMP_DIR=$(mktemp -d)
+            ALPHA="${TEMP_DIR}/alpha.png"
+            PROCESSED="${TEMP_DIR}/processed.png"
+
+            # Extract alpha channel
```

Comment:
With all these repetition, I suggest moving the code to a script which would be called here and there.

Or even define a local github "action"

Reply:
Could be. Hard to define inputs for it, though. The two variables are the relative path to the images, and find's regexp ignore pattern. Not very intuitive.

Reply:
The script is there, but for local use, so no `./` because of downloaded AppImage. Maybe I could reuse that.